### ISSUE TITLE
Unexpected DNS Lookup caused by RestSharp => move to `HttpClient`

### ISSUE THREAD

**Issue by SGStino** [influxdata/influxdb-client-csharp#225]
__Steps to reproduce:__
List the minimal actions needed to reproduce the behavior.

**Program.cs:**
```csharp
var client = InfluxDB.Client.InfluxDBClientFactory.Create(InfluxDBClientOptions.Builder.CreateNew()
    .Url("http://influx.server.svc.cluster.local:8086")
    .Bucket("test1")
    .Org("test1")
    .Build());

var api = client.GetWriteApiAsync();
int i = 0;
while (true)
{
    var point = PointData.Builder
        .Measurement("test")
        .Timestamp(DateTime.UtcNow, InfluxDB.Client.Api.Domain.WritePrecision.Ms)
        .Field("write", i++)
        .ToPointData();

    await api.WritePointsAsync(point);

    await Task.Delay(TimeSpan.FromMilliseconds(50));
}
```

__Expected behavior:__
I'd expect the client to do:
1. a DNS lookup
2. an HTTP/1.1 POST to /api/v2/signin
3. an HTTP/1.1 POST to /api/v2/write
4. repeat step 3 forever until:
    - login token expired (application error) -> step 2
    - endpoint unreachable (network error) -> step 1 (dns might have changed)

(and actually, skip 2 if there are no credentials provided, considering it's a no-op)

__Actual behavior:__

1. a DNS looup
2. an HTTP/1.1 POST to /api/v2/signin
3. a DNS lookup
4. an HTTP/1.1 POST to /api/v2/write
5. repeat from 1

![image](https://user-images.githubusercontent.com/5156315/127446963-471073c0-379a-4793-98ae-992d375b53b4.png)

Also, i'm noticing that the HTTP/1.1 POST to /api/v2/write has a connection keep-alive header. However, it's setting (SYN) up and tearing down (FIN) a new connection each time

__Specifications:__
 - Client Version: latest nuget as of writing (2.0.0, published 9th of july, 2021)
 - InfluxDB Version: 1.8
 - Platform: docker

**Dockerfile**
```Dockerfile
FROM mcr.microsoft.com/dotnet/runtime:5.0 AS base
WORKDIR /app

FROM mcr.microsoft.com/dotnet/sdk:5.0 AS build
WORKDIR /src
COPY ["InfluxClientAPITest/InfluxClientAPITest.csproj", "InfluxClientAPITest/"]
RUN dotnet restore "InfluxClientAPITest/InfluxClientAPITest.csproj"
COPY . .
WORKDIR "/src/InfluxClientAPITest"
RUN dotnet build "InfluxClientAPITest.csproj" -c Release -o /app/build

FROM build AS publish
RUN dotnet publish "InfluxClientAPITest.csproj" -c Release -o /app/publish

FROM base AS final
WORKDIR /app
COPY --from=publish /app/publish .
ENTRYPOINT ["dotnet", "InfluxClientAPITest.dll"]
```

**Comment by SGStino** [comment]
It might be related to the use of RestSharp:
```csharp
var client = new RestClient("http://influx.server.svc.cluster.local:8086/api/v2/write");

int i = 0;
while (true)
{
    var point = PointData.Builder
        .Measurement("test")
        .Timestamp(DateTime.UtcNow, InfluxDB.Client.Api.Domain.WritePrecision.Ms)
        .Field("write", i++)
        .ToPointData();


    var request = new RestRequest(Method.POST);
    request.AddQueryParameter("org", "test1");
    request.AddQueryParameter("bucket", "test1");
    request.AddQueryParameter("precision", "ms");
    request.AddParameter("text/plain", point.ToLineProtocol(), ParameterType.RequestBody);
    await client.ExecutePostAsync(request);

    await Task.Delay(TimeSpan.FromMilliseconds(50));
}
```
produces exactly the same issues: dns lookup, new tcp connection, ...

However, if i switch to .NET's HttpClient directly:

```csharp
var client = new HttpClient();

int i = 0;
while (true)
{
    var point = PointData.Builder
        .Measurement("test")
        .Timestamp(DateTime.UtcNow, InfluxDB.Client.Api.Domain.WritePrecision.Ms)
        .Field("write", i++)
        .ToPointData();


    var content = new StringContent(point.ToLineProtocol(), Encoding.UTF8, "text/plain");

    var result = await client.PostAsync("http://influx.server.svc.cluster.local:8086/api/v2/write?org=test1&bucket=test1&precision=ms", content);

    await Task.Delay(TimeSpan.FromMilliseconds(50));
}
```

then i get clean re-use of the TCP connection and a whole lot less load on the entire kubernetes cluster:
![image](https://user-images.githubusercontent.com/5156315/127449509-0b395925-2d15-4f58-b574-e7504e3bc455.png)

**Comment by bednar** [comment]
@SGStino, thanks for detail investigation. 

I think the problem could be also caused by unwanted `HTTP/1.1 POST to /api/v2/signin` operation. 

Could you please try to initialize the client with dummy token? Something like:

```csharp
var client = InfluxDB.Client.InfluxDBClientFactory.Create(InfluxDBClientOptions.Builder.CreateNew()
    .Url(MockServerUrl)
    .AuthenticateToken("test1-token")
    .Bucket("test1")
    .Org("test1")
    .Build());
```

Regards

**Comment by SGStino** [comment]
```csharp
var client = InfluxDB.Client.InfluxDBClientFactory.Create(InfluxDBClientOptions.Builder.CreateNew()
    .Url("http://influx.server.svc.cluster.local:8086")
    .AuthenticateToken("test")
    .Bucket("test1")
    .Org("test1")
    .Build());

var api = client.GetWriteApiAsync();
int i = 0;
while (true)
{
    var point = PointData.Builder
        .Measurement("test")
        .Timestamp(DateTime.UtcNow, InfluxDB.Client.Api.Domain.WritePrecision.Ms)
        .Field("write", i++)
        .ToPointData();


    await api.WritePointsAsync(point); 
    await Task.Delay(TimeSpan.FromMilliseconds(50));
}
```
That cuts the number of dns queries on half, yes :) 
But since the problem rests with RestSharp... it doesn't fix it entirely:
![image](https://user-images.githubusercontent.com/5156315/128060151-979f9114-634f-4411-9cfd-2a101db077b5.png)

**Comment by bednar** [comment]
Hi @SGStino, 

We are preparing a new major version of the client (`v4.0.0`) which already uses `HttpClient`. Please try the preview version by `dotnet add package InfluxDB.Client --version 4.0.0-dev.5084`. 

Any feedback will be appreciated 👂.

Regards

**Comment by bednar** [comment]
@SGStino, the issue is fixed in new major update of the client - [`v4.0.0`](https://github.com/influxdata/influxdb-client-csharp/milestone/29) - which contains a few breaking changes. The most important change is in underlaying HTTP client - switch from legacy `HttpWebRequest` class, and uses `HttpClient` instead.

Please try a preview version by `dotnet add package InfluxDB.Client --version 4.0.0-dev.5280`. 

Any feedback will be appreciated 👂. For more info about new major version see [CHANGELOG.md](https://github.com/influxdata/influxdb-client-csharp/blob/master/CHANGELOG.md#400-unreleased).

**Comment by SGStino** [comment]
Seems like the public API seems to have broken a little since 2.0.0. I found time to update our code, just haven't been able to run it in an environment where I can inspect DNS requests easily. (remote work doesn't make things easier)
But I suspect that it'll behave much like the raw HttpClient test by re-using open connections.

I believe a lot of CPUs will thank you for this change 👍
